awx/main/tests/functional/test_named_url.py

```diff
@@ -0,0 +1,244 @@
+import pytest
+
+from awx.api.versioning import reverse
+from awx.main.middleware import URLModificationMiddleware
+from awx.main.models import * # noqa
+
+
+@pytest.mark.django_db
+def test_user(get, admin_user):
+    # In real-world scenario, named url graph structure is populated by __init__
+    # of URLModificationMiddleware. The way Django bootstraps ensures the initialization
+    # will happen *once and only once*, while the number of initialization is uncontrollable
+    # in unit test environment. So it is wrapped by try-except block to mute any
+    # unwanted exceptions.
+    try:
+        URLModificationMiddleware()
+    except Exception:
```

Comment:
Is there a more reliable exception thrown here that's more specific than `Exception`?

Reply:
Instead of putting this block at the top of every method, you could define a local pytest fixture:

```python
@pytest.fixture(scope='function', autouse=True)
def init_url_modification_middleware():
    # In real-world scenario, ...
    try:
        URLModificationMiddleware()
    except:
        pass
```

Reply:
Good find. Autouse fixture drops redundancy a lot.